cmake/kokkos_arch.cmake

```diff
@@ -534,10 +534,11 @@ IF (KOKKOS_ENABLE_OPENMPTARGET)
   SET(CLANG_CUDA_ARCH ${KOKKOS_CUDA_ARCH_FLAG})
   IF (CLANG_CUDA_ARCH)
     STRING(REPLACE "sm_" "cc" NVHPC_CUDA_ARCH ${CLANG_CUDA_ARCH})
+# [DEBUG on Feb. 18, 2022 by Seyong Lee] Add -O3 option to both Clang and NVHPC.
     COMPILER_SPECIFIC_FLAGS(
-      Clang -Xopenmp-target -march=${CLANG_CUDA_ARCH} -fopenmp-targets=nvptx64-nvidia-cuda
+      Clang -O3 -Xopenmp-target -march=${CLANG_CUDA_ARCH} -fopenmp-targets=nvptx64-nvidia-cuda
```

Comment:
O3 option is added here so that it can be included as a default compiler option, since we noticed that there can be huge performance difference between with and without the O3 option when compiling the OpenMPTarget backend using Clang. We added the O3 option to the NVHPC OpenACC compiler for the same performance reason. If this violates the Kokkos implementation policy, we can remove it and let users add as necessary.

Reply:
By default, you have
- `CMAKE_CXX_FLAGS_DEBUG="-g"`
- `CMAKE_CXX_FLAGS_RELEASE="-O3 -DNDEBUG"`, 
- `CMAKE_CXX_FLAGS_RELWITHDEBINFO="-O2 -g -DNDEBUG"`

and we set use `CMAKE_BUILD_TYPE=RelWithDebInfo` by default. I don't see any reason to force something here.